src/libraries/System.Text.Json/gen/Helpers/KnownTypeSymbols.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
+
+namespace System.Text.Json.Reflection
+{
+    internal sealed class KnownTypeSymbols(Compilation compilation)
```

Comment:
cc @layomia this implements the pattern you've been using in the ConfigurationBinder source generator.